src/api/c/DependencyModule.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <DependencyModule.hpp>
+#include <algorithm>
+#include <string>
+
+using std::string;
+namespace {
+
+#if(OS_WIN)
+    static const char* librarySuffix = ".dll";
+    static const char* libraryPrefix = "";
+
+    LibHandle loadLibrary(const char* library_name) {
+        return LoadLibrary(library_name);
+    }
+
+    void unloadLibrary(LibHandle handle) {
+        FreeLibrary(handle);
+    }
+
+    string getErrorMessage() {
+        LPVOID lpMsgBuf;
+        LPVOID lpDisplayBuf;
+        DWORD dw = GetLastError();
+
+        size_t characters_in_message;
+        FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+                      FORMAT_MESSAGE_FROM_SYSTEM |
+                      FORMAT_MESSAGE_IGNORE_INSERTS,
+                      NULL,
+                      dw,
+                      MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                      (LPTSTR) &lpMsgBuf,
+                      0, NULL );
+        string error_message(lpMsgBuf);
+        return error_message;
+    }
+
+#else
+#if(OS_MAC)
+    static const char* librarySuffix = ".dylib";
+#else
+    static const char* librarySuffix = ".so";
+#endif
+
+    static const char* libraryPrefix = "lib";
+
+    LibHandle loadLibrary(const char* library_name) {
+        return dlopen(library_name, RTLD_LAZY);
+    }
```

Comment:
I think it would be easy to understand the code if the `#if` blocks based on OS are closer to the code deviation point as follows.
```
LibHandle loadLibrary(const char* library_name)
{
+#if(OS_WIN)
        return LoadLibrary(library_name);
#else
        return dlopen(library_name, RTLD_LAZY);
#endif
}
```
instead of current approach.  This wouldn't be a big distraction when the functions have single line implementations. However, tracking where one `#if` begins and ends is sometimes a distraction and can be confusing. It is hard to loose track when the matching `#if` and `#endif` are close by.

Reply:
Yeah, that's why I separated them. It's easier for me to follow what is going on. I am thinking about having one interface file and have os specific implementations in a separate file because I don't really like the current approach either.

Reply:
Moved the platform independent code in its own file. Its much easier to read. Let me know what you think.